src/Uno.UWP/Web/Http/CookieManager.cs

```diff
@@ -0,0 +1,95 @@
+﻿#if __WASM__
+#nullable enable
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Uno.Foundation;
+using Windows.Web.Http;
+
+namespace Uno.Web.Http
+{
+	public class CookieManager
+	{
+		private const char ValueSeparator = '=';
+		private const char AttributeSeparator = ';';
+
+		private static readonly Lazy<CookieManager> _cookieManager = new Lazy<CookieManager>(() => new CookieManager());
+
+		private CookieManager()
+		{
+		}
+
+		public static CookieManager GetDefault() => _cookieManager.Value;
+
+		public void SetCookie(SetCookieRequest cookie)
+		{
+			var httpCookie = new HttpCookie(cookie.Name, cookie.Domain ?? string.Empty, cookie.Path ?? string.Empty)
+			{
+				Secure = cookie.Secure,
+				Expires = cookie.Expires,
+				Value = cookie.Value,
+			};
+			var serializedCookie = httpCookie.ToString();
+
+			if (cookie.MaxAge != null)
+			{
+				serializedCookie += $"; max-age={cookie.MaxAge.Value.ToString(CultureInfo.InvariantCulture)}";
+			}
+			if (cookie.SameSite != null)
+			{
+				serializedCookie += $"; samesite={cookie.SameSite.Value.ToString("g").ToLowerInvariant()}";
+			}
+
+			var escapedCookie = WebAssemblyRuntime.EscapeJs(serializedCookie);
+			var jsInvoke = $"document.cookie = '{escapedCookie}'";
+			WebAssemblyRuntime.InvokeJS(jsInvoke);
+		}
+
+		public void DeleteCookie(string name, string? path = null)
+		{
+			var setCookieRequest = new SetCookieRequest(name, string.Empty)
+			{
+				Expires = DateTimeOffset.MinValue,
+				Path = path
+			};
+
+			SetCookie(setCookieRequest);
+		}
+
+		public Cookie? GetCookie(string name) => GetCookies().FirstOrDefault(c => c.Name == name);
```

Comment:
Lookups returning null are best named with Find instead of Get.

```suggestion
		public Cookie? FindCookie(string name) => GetCookies().FirstOrDefault(c => c.Name == name);
```